File: DKS-API/DTOs/SF340PPDSchedule.cs
using System;
using DKS_API.Helpers;

namespace DKS_API.DTOs
{
    public class SF340PPDSchedule : PaginationParams
    {
        public string factory { get; set; }
        public string season { get; set; }
        public string bpVer { get; set; }
        public string cwaDateS { get; set; }
        public string cwaDateE { get; set; }
        public string article { get; set; }
        public string modelNo { get; set; }
        public string modelName { get; set; }


    }
}

File: DKS-SPA/src/app/core/_models/s_f340-ppd-schedule.ts
import { Pagination } from "./pagination";
import { utilityConfig } from "../utility/utility-config";

export class SF340PpdSchedule extends Pagination {
  factory: string;
  season: string;
  bpVer: string;
  cwaDateS: string;
  cwaDateE: string;
  article: string;
  modelNo: string;
  modelName: string;
  ubType:string;
  /**
   *default set of searching parameters
   */
  constructor() {
    super();
    this.factory = "";
    this.season = "";
    this.bpVer = "";
    this.cwaDateS = "";
    this.cwaDateE = "";
    this.isPaging = true; //開分頁
    this.article = "";
    this.modelNo = "";
    this.modelName = "";
    this.ubType ="";
  }

  public setPagination(pagination: Pagination) {
    this.currentPage = pagination.currentPage;
    this.itemsPerPage = pagination.itemsPerPage;
    this.totalItems = pagination.totalItems;
    this.totalPages = pagination.totalPages;
  }
}
